src/hotspot/share/gc/g1/g1YoungCollector.cpp

```diff
@@ -79,10 +80,11 @@ class G1YoungGCTraceTime {
   const char* update_young_gc_name() {
     snprintf(_young_gc_name_data,
              MaxYoungGCNameLength,
-             "Pause Young (%s) (%s)%s",
+             "Pause Young (%s) (%s)%s%s",
              G1GCPauseTypeHelper::to_string(_pause_type),
              GCCause::to_string(_pause_cause),
-             _collector->evacuation_failed() ? " (Evacuation Failure)" : "");
+             _collector->evacuation_pinned() ? " (Pinned)" : "",
+             _collector->evacuation_alloc_failed() ? " (Allocation Failure)" : "");
```

Comment:
> GC(6) Pause Young (Normal) (Pinned) (Evacuation Failure)

I wonder if the last two can be merged into one `()`, sth like `(Pinned / ...)`, because they are on the same abstraction level.

Reply:
Parsing the separate components is easier :) Not sure if these tags in any way ever indicated some level of abstraction.

I do not have a strong opinion here. The combinations
```
(Pinned)
(Allocation Failure)
(Pinned + Allocation Failure)  // or the other way around, or some other symbol for "+" or no symbol at all?
```
are fine with me (and I thought about doing something more elaborate here), but my concern has been that any complicated string makes it less unique (e.g. `(Allocation Failure)` vs. "Allocation Failure") and adds code both to implement and parse the result.

Much more disrupting is likely that there is no "Evacuation Failure" string any more. But log messages are not part of the external interface, and we should not want to change them just because.

Reply:
The example looks good to me.

Reply:
Have the final output looking something like this was agreed on during internal discussion:
GC(6) Pause Young (Normal) (Evacuation Failure: Pinned) 1M->1M(22M) 36.16ms
GC(6) Pause Young (Normal) (Evacuation Failure: Allocation) 1M->1M(22M) 36.16ms
GC(6) Pause Young (Normal) (Evacuation Failure: Allocation / Pinned) 1M->1M(22M) 36.16ms

Reply:
Done.